plugins/score-card/src/components/EntityScoreCardTable/EntityScoreCardTable.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2022 Oriflame
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useEffect } from 'react';
+import { Progress } from '@backstage/core-components';
+import useAsync from 'react-use/lib/useAsync';
+import {
+  errorApiRef,
+  useApi,
+  githubAuthApiRef,
+} from '@backstage/core-plugin-api';
+import { getWarningPanel } from '../../helpers/getWarningPanel';
+import { scoringDataApiRef } from '../../api';
+import { ScoreTable } from '../ScoreCardTable/ScoreCardTable';
+import { useEntity } from '@backstage/plugin-catalog-react';
+
+const useScoringAllDataLoader = (entityKindFilter?: string[]) => {
```

Comment:
is there any filtering based on the entity the user is looking at? I have tried to add the `EntityScoreCardTable` under a group and it shows also other groups, i.e. the very same presentation/data, as the common scoreboard..

`packages/app/src/components/catalog/EntityPage.tsx`:
```typescript
const groupPage = (
  <EntityLayout>
...
        <EntityLayout.Route path="/scoreboard" title="Score board">
            <Grid container spacing={3} alignItems="stretch">
                <Grid item xs={12}>
                    <EntityScoreCardTable />
                </Grid>
            </Grid>
        </EntityLayout.Route>
  </EntityLayout>
);
```

![image](https://github.com/Oriflame/backstage-plugins/assets/16001792/416cfc59-4b8b-429e-a941-940457343db5)

I though that the main idea of this component is to provide a quick way to see "my" (group) entities and/or entities connected to the respective entity...e.g. scoring for all components under a system etc..

Reply:
> is there any filtering based on the entity the user is looking at? I have tried to add the `EntityScoreCardTable` under a group and it shows also other groups, i.e. the very same presentation/data, as the common scoreboard..
This UI is exactly same as `ScoreBoardPage`.

Reply:
> ![image](https://private-user-images.githubusercontent.com/16001792/286421552-416cfc59-4b8b-429e-a941-940457343db5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEyNTM5NDIsIm5iZiI6MTcwMTI1MzY0MiwicGF0aCI6Ii8xNjAwMTc5Mi8yODY0MjE1NTItNDE2Y2ZjNTktNGI4Yi00MjllLWE5NDEtOTQwNDU3MzQzZGI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI5VDEwMjcyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEzYjg0MjU4ZmU5OTM5NmY2MWEwMzk3YzhiY2YwZDlmYzlmNmE4ZWQ3NmIxMTI4ZTk4MTQ4M2NhMWI5MzI2OGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cJQP8haSxVKGfnv7_umx8k-3v9Pql0CWOrj7OONkl_s)
> 
> I though that the main idea of this component is to provide a quick way to see "my" (group) entities and/or entities connected to the respective entity...e.g. scoring for all components under a system etc..

`EntityScoreBoardTable` component carries entityRef, using which we could read the annotations of that entity. So if a repository has configured `scorecard/jsonDataUrl` annotation pointing to a JSON file, we can dynamically read that JSON and get the data. This will help as the JSON file can be dynamically generated by a project's repository through an automation. The project can contain a System, followed by sub components such as api/website/service etc. So System annotation can refer to `all.json` and each of the sub components can have their own `score.json`.

Reply:
Agree to your points. To summarize,

- We can get rid of `all.json`, and in an entity annotation we can decide if that is parent of other or not, like
```
annotations:    
    github.com/project-slug: anand-tyagaraj/score-card-samples        
    scorecard/jsonDataUrl: './samples/score.json'
    scorecard/children: 
      - org/score-card-child1/catalog-info.yaml
      - org/score-card-child2/catalog-info.yaml
```
- If that entity needs to show aggregate then, we would just need its children to dynamically build a tabular format
- By doing this we can avoid duplicate of maintaining `ScoreCardTable` and `EntityScoreBoardTable`
- So you recommend to completely move out of reading `scorecards\jsonDataURL` from `app-config.yaml`? Since if the proposal for `ScoreBoardPage` is to read all entities or an org and aggregate their score as table we would not need `all.json` at all right?

This looks like bit of tearing up and refactoring, I can plan this.

Reply:
Thanks a lot! Just a little adjustments:
- instead of `annotations['scorecard/children']` we can use native Backstage API to obtain descendants of that entity.
- we shall still support `all.json` from a `scorecards\jsonDataURL` setting in `app-config.yaml` for having a single source of all scoring data (and to keep backward compatibility for those, who has it implemented like this.

It would maybe help to split this PR into the following tasks (new PRs? keep it as is in a single PR?):
1. add `entity?: Entity` to `getAllScores` in `ScoringDataApi` and implement it in `ScoringDataJsonClient` in a way, that in case the entity is provided, it will return scoring data for the entity (and it's children) only (from `all.json`. This is a non-breaking change.
2.  `ScoreCardTable` -> `EntityScoreBoardTable`. Adds support for obtaining the current entity (via `useEntity()`) and wil pass it to underlayng ScoringDataApi
3. Support for reading the `scorecard/jsonDataUrl` annotation from the entities. Add the `entity?: Entity` also to the `getScore` in `ScoringDataApi`: if we pass the entity and it's annotated by `scorecard/jsonDataUrl`, we will read the scoring from provided url. If it's not annotatea, previous behaviour should apply = we will use the `scorecards\jsonDataURL` setting in `app-config.yaml`. 
4. (definitely a new PR, WDYT?) Introduce a wrapper for `EntityPage` view (for the `/scoreboard` route): if the `getAllScores(entity)` returns more than a single score, we will show the `EntityScoreBoardTable`. If only a single score is returned, we will show `EntityScoreCardContent`.  We should probably have a condition for showing the wrapper in case no score is returned. We should also have the `EntityScoreCardContent` integrated in the EntityPage view for the `/score` route (to display a single score even for a parent entity if this exists)

Reply:
Yes, we can split PRs as below

PR 1: Item 2 and  3
PR 2: Item 1
PR 3: Item 4?

Reply:
would be great, thanks!